### ISSUE TITLE
Recipe conflicts? Between _masonry and _gates.

### ISSUE THREAD

**Issue by enkidu-smith** [minetest-mods/castle_masonry#5]
I'm trying to create some arrow slits, using the recipe that has two vertical rows of material. I was actually using stone brick blocks and the recipe results in an "Unknown item". Looking at my player file what is actually created is a block called "castle_gates:stonebrick_portcullis_slot". Looking deeper both that item and "castle_masonry:arrowslit_stonebrick" use the same recipe, eg 
```
                recipe = {
                {material.craft_material,"", material.craft_material},
                {material.craft_material,"", material.craft_material},
                {material.craft_material,"", material.craft_material} },

```
I don't know if this is significant as I don't know how minetest handles such things and I don't know if this relates to the "Unknown item" issue. Can someone please advise on this issue? Thanks, Cliff

**Comment by enkidu-smith** [comment]
OK, I'm no programmer, but looking at the code for castle_gates it seemed to me that the section of code for portcullis was not needed (or referenced anywhere else AFAICT). I commented it out and the arrow slits stuff started working. I'm not confident enough to create a patch/pull request.

Commented out code in castle/castle_gates/gate_slots.lua is as follows
 
```
        -- minetest.register_craft({
        -- output = mod_name..":"..material.name.."_portcullis_slot 3",
        -- recipe = {
        --      {material.craft_material,"",material.craft_material},
        --      {material.craft_material,"",material.craft_material},
        --      {material.craft_material,"",material.craft_material},
        --      },
        -- })

        -- minetest.register_craft({
        --      output = mod_name..":"..material.name.."_portcullis_slot",
        --      type = "shapeless",
        --      recipe = {mod_name..":"..material.name.."_portcullis_slot_reverse"},
        -- })
        -- minetest.register_craft({
        --      output = mod_name..":"..material.name.."_portcullis_slot_reverse",
        --      type = "shapeless",
        --      recipe = {mod_name..":"..material.name.."_portcullis_slot"},
        -- })

        -- if burn_time > 0 then
        --      minetest.register_craft({
        --              type = "fuel",
        --              recipe = mod_name..":"..material.name.."_portcullis_slot",
        --              burntime = burn_time * 2,
        --      })
        --      minetest.register_craft({
        --              type = "fuel",
        --              recipe = mod_name..":"..material.name.."_portcullis_slot_reverse",
        --              burntime = burn_time * 2,
        --      })
        -- end

```

**Comment by enkidu-smith** [comment]
Oops! Didn't mean to close it.

**Comment by enkidu-smith** [comment]
Raised issue in [castle_gates](https://github.com/minetest-mods/castle_gates/issues/4).

**Comment by FaceDeer** [comment]
Sorry about taking several months to get to this. I went on a bit of a modding hiatus while real life got busy. Back now, and boy is this an embarrassing bug - the mod actually defines "gate_slot" and "gate_slot_reverse" blocks, designed to allow gates to fit more snugly against a wall or to allow a portcullis to slide through a space without allowing a player to fit through it. But I used the wrong name in the recipe.

I've also modified the recipe so it doesn't collide with the arrow slits. Also an embarrassing bug, since I was working on both castle_gates and castle_masonry simultaneously when I did this and should have noticed.
